src/Linux/curl_easy.cpp

```diff
@@ -136,7 +144,21 @@ curl_easy::~curl_easy()
 
 void curl_easy::perform() const
 {
-    CURLcode result = curl_easy_perform(handle);
+	bool finished = false;
+	int retry_delay = initial_retry_delay_ms;
+	CURLcode result;
+	
+	for(int i = 0; i < maximum_retries && !finished; i++){
+		result = curl_easy_perform(handle);
+		
+		if(result == CURLE_OPERATION_TIMEDOUT){
+			std::this_thread::sleep_for(std::chrono::milliseconds(retry_delay));
+			retry_delay *= 2;

```

Comment:
Is 20 ms even worthwhile to delay? I.e What use case would resolve in 20 ms or even at the best in this case 320 ms?

Reply:
We only do this delay in case of ERROR_WINHTTP_TIMEOUT or ERROR_WINHTTP_RESEND_REQUEST. In these cases we already have waited for a bit or have a confirmation that we want to retry as soon as possible.